tests/trace_parser/trace_parser.cpp

```diff
@@ -0,0 +1,188 @@
+#include "trace_parser.h"
+
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <string>
+
+namespace m3 {
```

Comment:
Can we indentation for the namespace?
Also, start curly braces from the new line